clients/drcachesim/tools/opcode_mix.cpp

```diff
@@ -229,6 +229,20 @@ opcode_mix_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
         instr_free(dcontext_.dcontext, &instr);
     }
     ++shard->opcode_counts[opcode];
+
+    uint category;
+    instr_t instr;
+    instr_init(dcontext_.dcontext, &instr);
+    app_pc next_pc = decode_from_copy(dcontext_.dcontext, decode_pc, trace_pc, &instr);
```

Comment:
Decoding is expensive and we took some effort above to cache the results, so let's leverage the same caching by using the same decode call as above and storing the categories in the (maybe renamed) opcode_cache.

Reply:
Done.